### ISSUE TITLE
Code implementations of model inference

### ISSUE THREAD

**Issue by Sanzhen3** [sunlicai/HiCMAE#7]
Thank you for your great work!
I wonder that how could I find the code implementations of model inference on a newly unseen dataset.

Looking forward to your reply,
Thank you.

**Comment by sunlicai** [comment]
Hi, thanks for your interest in our work!

For a new dataset, you should prepare the data (see Data Preparation) and register the dataset in dataset_av.py.

**Comment by Sanzhen3** [comment]
Thank you for your quick reply!
What I mean is how to load the weights of the trained model and purely perform model inference. Could you provide code implementations related to the inference or indicate how I should proceed?

Thanks a lot.

**Comment by Sanzhen3** [comment]
> Thank you for your quick reply! What I mean is how to load the weights of the trained model and purely perform model inference. Could you provide code implementations related to the inference or indicate how I should proceed?
> 
> Thanks a lot.

For this issue, in other words, I just want to load the previously trained checkpoint on the downstream dataset and conduct evaluations on the **test** set of the specific downstream dataset.

I have carefully read the log of [**ferv39k.out**](https://github.com/sunlicai/MAE-DFER/tree/master/logs) which is provided in your previous work **MAE-DFER**. I find that **_there are only model running on the test set_**, which meet my requirements. Furthermore, I notice that there are
`Auto resume checkpoint: saved/model/finetuning/ferv39k/voxceleb2_videomae_pretrain_base_dim512_local_global_attn_depth16_region_size2510_patch16_160_frame_16x4_tube_mask_ratio_0.9_e100_with_diff_target_server170/checkpoint-49/eval_lr_1e-3_epoch_100_size160_sr1_classify_token_type_region_server164/checkpoint-99.pth`

`Resume checkpoint saved/model/finetuning/ferv39k/voxceleb2_videomae_pretrain_base_dim512_local_global_attn_depth16_region_size2510_patch16_160_frame_16x4_tube_mask_ratio_0.9_e100_with_diff_target_server170/checkpoint-49/eval_lr_1e-3_epoch_100_size160_sr1_classify_token_type_region_server164/checkpoint-99.pth` in Line 1342 and 1343 of the log.

As a result, I have tried my best to find the locations of **auto resume checkpoint** and **resume checkpoint**, but I do not find them.

Therefore, could you help me to figure out how to set **auto resume checkpoint** and **resume checkpoint** so that I can load the model and evaluate on the **test** set?

Any reply will be appreciated greatly.
Thank you very much.

**Comment by sunlicai** [comment]
Sorry for late reply! 

For now, we do not support sample-level inference. However, you can test your fine-tuned model on customized videos in batch level by using '--eval' option in the command line. Note that you should prepare your videos in a csv file (i.e., video path + label) for testing in a similar way with fine-tuning. The checkpoint can be automatically loaded by specifing the checkpoint path in the '--finetune' or '--resume' option.

**Comment by Sanzhen3** [comment]
> Sorry for late reply!
> 
> For now, we do not support sample-level inference. However, you can test your fine-tuned model on customized videos in batch level by using '--eval' option in the command line. Note that you should prepare your videos in a csv file (i.e., video path + label) for testing in a similar way with fine-tuning. The checkpoint can be automatically loaded by specifing the checkpoint path in the '--finetune' or '--resume' option.

Thank you for your reply! I will have a try.

**Comment by Sanzhen3** [comment]
Nice try! My question is solved. Thank you for your help again!
